common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -284,11 +284,7 @@ export class ActiveKey implements LayoutKey {
       if(!keyboard.definesPositionalOrMnemonic) {
         // Not the best pattern, but currently safe - we don't look up any properties of any of the
         // arguments in this use case, and the object's scope is extremely limited.
-        Lkc.Lcode = KeyMapping._USKeyCodeToCharCode(keyboard.constructKeyEvent(null, null, {
-          K_CAPS: false,
-          K_NUMLOCK: false,
-          K_SCROLL: false
-        }));
+        Lkc.Lcode = KeyMapping._USKeyCodeToCharCode(keyboard.constructNullKeyEvent(null, null));
```

Comment:
Can you explain the reason for this change? I don't understand it

Reply:
Hmm... it _is_ worth noting that the original line doesn't reference an actual key.  Personally, when running a trace on what's going on, this block is extremely illuminating:

https://github.com/keymanapp/keyman/blob/4c7b7992da6491dc5bf30245dc5851f8c1d471c1/common/web/keyboard-processor/src/keyboards/keyboard.ts#L471-L473

Note that we're within the `constructBaseKeyEvent` method (for this comment).  L473 above is relying upon the results of said method.  This is an obvious problem.

Since this hasn't exactly been a section of code I've actually dived into much, perhaps a comparison against stable-10.0 is in order.  I've moved the code a few times since then, so I'll highlight the corresponding section to move things along:

https://github.com/keymanapp/keyman/blob/f16872a99896a2fc208c68ac92a827820ab07e3f/web/source/kmwdomevents.ts#L722-L734

(Note that this is constructing the event in full from nothing, rather than the partial construction intended for `constructBaseKeyEvent`.)

So... KMW 1.0 used different key-code values, and we just need to map the key code we use now to the one that was used then.  It's not even like we're replacing the other key-event values... just `.Lcode` here and `.LisVirtualKey` on the next line.

Something far closer to the original would likely be best here.

Reply:
> this block is extremely illuminating

Not to me! Perhaps you could explain it more clearly for me?

Reply:
> Note that we're within the constructBaseKeyEvent method (for this comment). L473 above is relying upon the results of said method. This is an obvious problem.

In other words, we have an impossible-to-meet precondition here.  We're within a method that must complete to satisfy the precondition of a method it's calling.

Reply:
Fortunately, we can simplify things greatly here... that's what the rest of the analysis is about.